packages/@tailwindcss-vite/src/index.ts

```diff
@@ -239,106 +266,57 @@ export default function tailwindcss(): Plugin[] {
-          // We expand string configs to their PostCSS config object similar to
-          // how Vite does it.
-          // See: https://github.com/vitejs/vite/blob/440783953a55c6c63cd09ec8d13728dc4693073d/packages/vite/src/node/plugins/css.ts#L1580
-          let searchPath = typeof postcssConfig === 'string' ? postcssConfig : config.root
-          let parsedConfig = await postcssrc({}, searchPath).catch((e: Error) => {
-            if (!e.message.includes('No PostCSS Config found')) {
-              if (e instanceof Error) {
-                let { name, message, stack } = e
-                e.name = 'Failed to load PostCSS config'
-                e.message = `Failed to load PostCSS config (searchPath: ${searchPath}): [${name}] ${message}\n${stack}`
-                e.stack = '' // add stack to message to retain stack
-                throw e
-              } else {
-                throw new Error(`Failed to load PostCSS config: ${e}`)
-              }
-            }
-            return null
-          })
-          if (parsedConfig !== null) {
-            postcssConfig = {
-              options: parsedConfig.options,
-              plugins: parsedConfig.plugins,
-            } as any
-          } else {
-            postcssConfig = {}
-          }
-          config.css = { postcss: postcssConfig }
-        }
-
-        // postcssConfig is no longer a string after the above. This test is to
-        // avoid TypeScript errors below.
-        if (typeof postcssConfig === 'string') {
-          return
-        }
-
-        if (!postcssConfig || !postcssConfig?.plugins) {
-          config.css = config.css || {}
-          config.css.postcss = postcssConfig || {}
-          config.css.postcss.plugins = [fixRelativePathsPlugin() as any]
-        } else {
-          postcssConfig.plugins.push(fixRelativePathsPlugin() as any)
-        }
-      },
-
       // Scan index.html for candidates
-      transformIndexHtml(html) {
-        let updated = scan(html, 'html')
+      transformIndexHtml(html, context) {
+        onFileChange(context.filename)
 
-        // In serve mode, if the generated CSS contains a URL that causes the
-        // browser to load a page (e.g. an URL to a missing image), triggering a
-        // CSS update will cause an infinite loop. We only trigger if the
-        // candidates have been updated.
+        let updated = scan(html, 'html')
         if (updated) {
-          updateCssModules(isSSR)
+          queueRebuild(isSSR)
         }
       },
 
       // Scan all non-CSS files for candidates
       transform(src, id, options) {
+        onFileChange(id)
+
         if (id.includes('/.vite/')) return
         let extension = getExtension(id)
         if (extension === '' || extension === 'css') return
 
-        scan(src, extension)
-        updateCssModules(options?.ssr ?? false)
+        let updated = scan(src, extension)
+        if (updated) {
+          queueRebuild(options?.ssr ?? false)
+        }
```

Comment:
Unclear why this was always causing css files to be rebuild. Changed it to only do that when new candidates are detected.